OKlib.js

```diff
@@ -824,416 +767,378 @@ registerPlugin({
+     * @param {Object[]} array
+     * @param {Object[]} elements
+     * @return {Object[]} returns a new array containing all elements of the two given arrays
      */
     function arrayCombineArrays(array, elements) {
         array = arrayCreateArray(array);
         elements = arrayCreateArray(elements);
         let result = [];
-        for (let i = 0; i < array.length; i++) {
+        for (let i in array) {
             result.push(array[i]);
         }
-        for (let j = 0; j < elements.length; j++) {
+        for (let j in elements) {
             result.push(elements[j]);
         }
-        log("arrayCombineArrays: Found '" + result.length + "' Objects", 4);
+        log('arrayCombineArrays: Found ' + result.length + ' Objects', 4);
         return result;
     }
 
     /**
-     * Checks if an Array contains the given Element
-     * @param  {Object[]} array
-     * @param  {Object} element
-     * @param  {Function} compare A Compare Function that should be used for the Comparison, if not set 'equal' is used
-     * @return {Boolean}  Returns true if the Element is contained in the Array
+     * checks if an array contains the given element
+     * @param {Object[]} array
+     * @param {Object} element
+     * @param {Function} compare a compare function that should be used for the comparison | if not set, 'equal' is used
+     * @return {Boolean} returns true if the element is contained in the array
      */
     function arrayContainsElement(array, element, compare) {
-        if (!compare) {
-            compare = equal;
-        }
-        for (let i = 0; i < array.length; i++) {
-            if (compare(array[i], element)) {
-                return true;
-            }
+        if (!compare) compare = equal;
+        for (let i in array) {
+            if (compare(array[i], element)) return true;
         }
         return false;
     }
 
     /**
-     * Checks if an Array contains at least one Element of another given Array
-     * @param  {Object[]} array
-     * @param  {Object[]} elements The Elements to search for in the Array
-     * @param  {Function} compare  A Compare Function that should be used for the Comparison, if not set 'equal' is used
-     * @return {Boolean}  Returns true if at least one Element is contained in the given Array
+     * checks if an array contains at least one element of another given array
+     * @param {Object[]} array
+     * @param {Object[]} elements the elements to search for in the array
+     * @param {Function} compare a compare function that should be used for the comparison | if not set, 'equal' is used
+     * @return {Boolean} returns true if at least one element is contained in the given array
      */
     function arrayContainsOne(array, elements, compare) {
-        for (let i = 0; i < elements.length; i++) {
-            if (arrayContainsElement(array, elements[i], compare)) {
-                return true;
-            }
+        for (let i in elements) {
```

Comment:
change this to for..of or revert to normal for loop, because for..in is not safe for array operation (might not iterate in the right order)